src/Core/src/Platform/iOS/MauiCheckBox.cs

```diff
@@ -24,7 +24,13 @@ public class MauiCheckBox : UIButton
 		bool _isEnabled;
 		bool _disposed;
 
-		public EventHandler? CheckedChanged;
+		readonly WeakEventManager _weakEventManager = new WeakEventManager();
+
+		public event EventHandler CheckedChanged
```

Comment:
I don't think the analyzer (yet) can tell this uses `WeakEventManager`. Can you add:
```suggestion
        [UnconditionalSuppressMessage("Memory", "MA0001", Justification = "Proven safe in test: CheckBoxTests.DoesNotLeak")]
		public event EventHandler CheckedChanged
```

```
src/Core/src/Platform/iOS/MauiCheckBox.cs(27,24): error MA0002: Member 'CheckedChanged' could cause memory leaks in an NSObject subclass. Remove the member, store the value as a WeakReference, or add the [UnconditionalSuppressMessage("Memory", "MA0002")] attribute with a justification as to why the member will not leak.
```

Reply:
Does making this weak cause the subscription in the handler to get collected prematurely?  If you force a GC does the CheckChanged event still fire? 

What if you subscribed to this using a static delegate? would that subscription survive a GC?

Reply:
Added